PNA.mdk

```diff
@@ -187,33 +187,41 @@ multiple NIC devices that are conformant to the PNA[^PNAPortability].
     programs that are able to compile on multiple NIC devices should
     process packets as described in this document.
 
-The Portable NIC Architecture (PNA) Model has four programmable P4
+The Portable NIC Architecture (PNA) Model has three programmable P4
 blocks and several fixed-function blocks, as shown in Figure
 [#fig-nic]. The behavior of the programmable blocks is specified using
-P4. The network ports, packet queues, and (optional) inline extern
-blocks are fixed-function blocks that can be configured by the control
+P4. The network ports, packet queues, and (optional) inline accelerators
+are fixed-function blocks that can be configured by the control
 plane, but are not intended to be programmed using P4.
 
 ~ Figure {#fig-nic; caption: "Programmable NIC Architecture Block Diagram"; page-align: here;}
 ![nic]
 ~
-[nic]: figs/pna-block-diagram.png { width: 100%;}
+[nic]: figs/flex-routing-arch.png { width: 100%;}
 
 ## Packet processing in the network to host direction
 
-Packets arriving from a network port first go through a `MainParser`,
+Packets arriving from a network port first go through a `Parser`,
 which is responsible for extracting all relevant packet headers.  The
-`MainControl` is typically where the bulk of code would be written for
+`Control` is typically where the bulk of code would be written for
 processing packets. It transforms headers, updates stateful elements
 like counters, meters, and registers, and optionally associates
-additional user-defined metadata with the packet. The `MainDeparser`
+additional user-defined metadata with the packet. The `Deparser`
 serializes the headers back into a packet that can be sent onwards.
 
-After the `MainDeparser`, the packet may either:
-* proceed to the message processing part of the NIC, and then typically on
+After the `Deparser`, the packet is processed by one or more
+inline accelerators. The P4 program executed in the `Control` determines
+whether and how each inline accelerator processes the packet by either setting
+corresponding intrinsic metadata or executing methods of corresponding extern
+object(s).
+Upon existing the inline accelerators, a packet may either:
```

Comment:
typo?  existing -> exiting?   Or perhaps another phrase such as "Upon completion of processing in the inline accelerators, ..."

Reply:
Fixed in latest commits